### PR TITLE
fix(normalize): 3'-align insertion→duplication; guarantee normalization idempotency

### LINKED ISSUE #864: test(normalize): guarantee normalization idempotency (normalize∘normalize == normalize)
`normalize` is single-pass (no whole-edit fixed-point loop). Surfaced during #852: a projected genomic *insertion* routes through `insertion_to_repeat` (which reports the 5'-most tract phase) and stops, but a *second* pass routes the resulting `Repeat` through `normalize_repeat` (which applies the 3'-shift) — so `normalize(normalize(x)) != normalize(x)` for that class. The CLI normalizes once (wrong), the conformance harness normalized twice (masked it).

#852 fixes the specific `insertion_to_repeat`/`normalize_repeat` pair, but other non-idempotent pairs may exist. Add a corpus-wide property test asserting `normalize(normalize(x)) == normalize(x)` across the grammar/spec and conformance corpora, and fix (or explicitly document) any cases it surfaces.

Relates to #852.

### PR DESCRIPTION
**Pull request by nh13**
Stacked on #869.

`normalize` is single-pass. #852 fixed one class where one pass produced a non-3'-most form a second pass would shift (the repeat path). This finds and fixes the **insertion→duplication** analogue, and adds property tests that lock `normalize(normalize(x)) == normalize(x)` corpus-wide.

## The bug

`insertion_to_duplication` 3'-aligned a single-copy tandem insertion only within the abutting rotation's phase. When an odd-length di/tri-repeat run continues one base further in the opposite phase (e.g. inserting `TG` at the 5' edge of a `TGTGTGTGT` run), the dup stopped one base 5' of the run's true 3' end — violating the HGVS 3'rule and producing a non-idempotent result. Pre-existing (reproduces before #852/#869); the CLI emits the non-canonical form.

## The fix

Apply `three_prime_align_tract` (the shared #852 helper) to the dup result for `ThreePrime`. Its loop `ref[end] == rotated[0]` is exactly the duplication slide rule `ref[s] == ref[s+L]`, so it slides the dup to the run's true 3'-most position and rotates the unit. `FivePrime` (#340) is unchanged.

Verified four ways that the most-3' form is correct: HGVS DNA `duplication.md` 3'rule; the slide-rule identity; **mutalyzer** on a real locus (`NC_000001.11:g.5010037_5010038insTG` → `g.5010045_5010046dup`, and mutalyzer re-normalizes the old phase-aligned form to it); and ferro+fix on the real reference. Red→green on that locus: pre-fix `g.5010044_5010045dup` → post-fix `g.5010045_5010046dup`.

## Idempotency guards (new)

- Manifest-gated conformance double-normalize sweep on both axes (`axis_normalized_idempotent`, `axis_genomic_idempotent`) — a hard invariant, no baseline ledger.
- CI-runnable reference-free corpus sweep over the HGVS-spec + edge-case inputs.
- CI-runnable regression lock for the off-phase insertion→dup class (value-pinned + idempotent), plus a `rules.rs` unit test.

## Impact

Updates 3 tests that had encoded the under-shifted (spec-violating) position. **Zero conformance-corpus change** (genomic ledger byte-identical; normalized axis 0 FAIL). Spawned #882 (a distinct, pre-existing out-of-phase false-dup bug found while cross-checking mutalyzer — not addressed here).

Closes #864

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved normalization to choose the true 3′-most duplication position for off-phase insertion scenarios, including partial-flank sliding behavior.
  * Corrected expected outputs for cyclic-rotation single-copy insertion cases so results now reflect the most-3′ aligned duplication.
  * Added regression coverage for a tricky di-repeat/off-phase repeat edge case to prevent early stopping.

* **Tests**
  * Expanded corpus idempotency coverage for normalization, including manifest-gated scenarios.
  * Added focused regression tests verifying stable repeated normalization for off-phase insertion→duplication cases.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

File: tests/it/idempotency_tests.rs [hunk 1]
```diff
@@ -565,3 +565,114 @@ fn test_clinvar_sample_parsing_idempotency() {
...
+    let mut inputs = load_spec_examples();
+    inputs.extend(load_edge_cases());
+
+    let mut tested = 0usize;
+    let mut transformed = 0usize;
+    let mut failures = Vec::new();
+
+    for input in &inputs {
+        let Ok(v) = parse_hgvs(input) else { continue };
+        let Ok(n1) = Normalizer::new(provider.clone()).normalize(&v) else {
```
[the remark this answers is not in the document]
Comment (nh13) [reply tests/it/idempotency_tests.rs:593]: Declining — the premise that these corpora are parse-clean does not hold empirically. Recording parse failures asserts **334** failures: the corpora carry valid-HGVS-but-not-ferro-parseable inputs, including HTML extraction noise (`class="invalid">NM_000109.3:c.-401C>T</code>`), translocations (`chr11:g.pter_15,825,272::chr2:...`), mosaic/chimeric notation (`LRG_199t1:r.(=/6_8del)`), and bare positions (`:c.11T>G`, `NM_004006.3:r.761`). The `valid` flag on edge cases means "valid HGVS per spec," not "ferro-parseable," and the spec-fixture rows include forms ferro intentionally does not accept. So the parse-failure `continue` is intentional (distinct from the re-parse/re-normalize `failures.push` paths, which catch round-trip regressions on inputs that *did* parse). The sweep still exercises 925 inputs, well above the `tested > 100` floor that guards against sample collapse.

**Comment by nh13** [comment]
@coderabbitai resume

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review

**Comment by nh13** [comment]
@coderabbitai review
